registry/session.go

```diff
@@ -281,7 +281,11 @@ func (r *Session) GetRepositoryData(remote string) (*RepositoryData, error) {
 	// TODO: Right now we're ignoring checksums in the response body.
 	// In the future, we need to use them to check image validity.
 	if res.StatusCode != 200 {
-		return nil, utils.NewHTTPRequestError(fmt.Sprintf("HTTP code: %d", res.StatusCode), res)
+		errBody, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, err
```

Comment:
This will hide the original application error is there is a problem reading the body. This should still return a `utils.NewHTTPRequestError` error, but perhaps without the body.

There are lots of cases where you get a non-200 response with an empty body. This code will fail to report that in a meaningful way to the user.